test/spec/EditorCommandHandlers-test.js

```diff
@@ -466,9 +489,314 @@ define(function (require, exports, module) {
+                lines[1] = "//    function bar() {";
+                lines[2] = "//        ";
+                lines[3] = "//        a();";
+                var expectedText = lines.join("\n");
+
+                testToggleLine(expectedText, {start: {line: 1, ch: 8}, end: {line: 3, ch: 11}}, "nction bar() {\n//        \n//        a");
+
+            });
+
+            it("should comment/uncomment when selection starts & ends on whitespace lines", function () {
+
+                myEditor.setSelection({line: 2, ch: 0}, {line: 4, ch: 8});
+
+                var lines = defaultContent.split("\n");
+                lines[2] = "//        ";
+                lines[3] = "//        a();";
+                lines[4] = "//        ";
+                var expectedText = lines.join("\n");
+
+                testToggleLine(expectedText, {start: {line: 2, ch: 0}, end: {line: 4, ch: 10}});
+
+            });
+
+            it("should comment/uncomment lines that were partially commented out already, our style", function () {
+                 // Start with line 3 commented out, with "//" at column 0
+                var lines = defaultContent.split("\n");
+                lines[3] = "//        a();";
+                var startingContent = lines.join("\n");
+                myDocument.setText(startingContent);
+
+                // select lines 1-3
+                myEditor.setSelection({line: 1, ch: 0}, {line: 4, ch: 0});
+
+                lines = defaultContent.split("\n");
+                lines[1] = "//    function bar() {";
+                lines[2] = "//        ";
+                lines[3] = "////        a();";
+                var expectedText = lines.join("\n");
+
+                testToggleLine(expectedText, {start: {line: 1, ch: 0}, end: {line: 4, ch: 0}});
+
+            });
+
+            it("should comment/uncomment lines that were partially commented out already, comment closer to code", function () {
+
+                // Start with line 3 commented out, with "//" snug against the code
+                var lines = defaultContent.split("\n");
+                lines[3] = "        //a();";
+                var startingContent = lines.join("\n");
+                myDocument.setText(startingContent);
+
+                // select lines 1-3
+                myEditor.setSelection({line: 1, ch: 0}, {line: 4, ch: 0});
+
+                lines = defaultContent.split("\n");
+                lines[1] = "//    function bar() {";
+                lines[2] = "//        ";
+                lines[3] = "//        //a();";
+                var expectedText = lines.join("\n");
+
+                testToggleLine(expectedText, {start: {line: 1, ch: 0}, end: {line: 4, ch: 0}});
+
+            });
+
+            describe("with multiple selections", function () {
+
+                it("should toggle comments on separate lines with cursor selections", function () {
+
+                    myEditor.setSelections([{start: {line: 1, ch: 4}, end: {line: 1, ch: 4}},
```

Comment:
@petetnt - again , could you give a pointer as to how its done , the tests are written like this in master since day 1 so have not got any idea about this particular style nit